### PR TITLE
Port require

### PR DESCRIPTION
**Pull request by shaleh**
Also port un_autoload.
Make Qnil and friends const instead of static which allows them in
turn to be used in other statics.

### CODE REVIEW

**Comment by shaleh** [comment]
Well this is annoying. It runs here.

File: src/fns.c [hunk 1]
```diff
@@ -3591,9 +3471,6 @@ compilation.  */);
   staticpro (&string_char_byte_cache_string);
   string_char_byte_cache_string = Qnil;
 
-  require_nesting_list = Qnil;
-  staticpro (&require_nesting_list);
```
Comment (Wilfred) [inline src/fns.c]: You don't seem to be protecting the variable from GC on the Rust side.

I've found some docs here: https://www.gnu.org/software/emacs/manual/html_node/elisp/Writing-Emacs-Primitives.html

Comment (shaleh) [reply src/fns.c]: Aye. This probably explains the failures in CI that I do not see locally.

I am preparing a PR with a rewrite of the build.rs that will also handle protected statics. The current build code shows my lack of experience coding in Rust. Once that PR is blessed I will update this PR.

**Comment by shaleh** [comment]
Waiting on #641.

File: src/fns.c (hunk 1 above)
Comment (Wilfred) [reply src/fns.c]: I did see some interesting bugs when I prototyped porting `cons` (which I still need to fix). If you deliberately create a some large lists in a loop you can trigger a few GCs to flush out issues.

Comment (shaleh) [reply src/fns.c]: Good suggestion. A similar large list is what is holding up the map PR.

**Pull request merged**